File: frontend/src/services/orderService.js
import api from "./api";

const orderService = {
  placeOrder: async (orderData) => {
    const { data } = await api.post("/orders", orderData);
    return data.data;
  },

  getMyOrders: async () => {
    const { data } = await api.get("/orders/my");
    return data.data;
  },

  getOrder: async (orderId) => {
    const { data } = await api.get(`/orders/${orderId}`);
    return data.data;
  },

  updateOrderStatus: async (orderId, status) => {
    const { data } = await api.put(
      `/orders/${orderId}/status?status=${status}`
    );

    return data.data;
  },

  deleteOrder: async (orderId) => {
    const { data } = await api.delete(`/orders/${orderId}`);
    return data.data;
  },
};

export default orderService;

File: backend/src/main/java/com/fitmart/backend/service/OrderService.java
package com.fitmart.backend.service;

import com.fitmart.backend.dto.request.PlaceOrderRequest;
import com.fitmart.backend.dto.response.OrderResponse;
import com.fitmart.backend.enums.OrderStatus;

import java.util.List;

public interface OrderService {

    OrderResponse placeOrder(
            String email,
            PlaceOrderRequest request);

    List<OrderResponse> getMyOrders(String email);

    OrderResponse getOrderById(
            String email,
            Long orderId);

    OrderResponse updateStatus(
            Long orderId,
            OrderStatus status);

    void deleteOrder(Long orderId);

}